src/util/recorder.ts

```diff
@@ -1751,6 +1751,13 @@ function createResponse(
 
   const httpHeaders = reqresp.getResponseHeadersDict(reqresp.payload.length);
 
+  const cookie =
+    reqresp.requestHeaders &&
+    (reqresp.requestHeaders["cookie"] || reqresp.requestHeaders["Cookie"]);
+  if (cookie) {
+    httpHeaders["x-wabac-preset-cookie"] = cookie;
+  }
+
```

Comment:
I think unlike the implementation in AWP, `cookie` here is just a boolean because we're testing whether reqresp.requestHeaders["cookie"] or reqresp.requestHeaders["Cookie"] exist but never actually setting `cookie` as their value.

Instead we probably want to do something like:

```suggestion
  const hasCookie =
    reqresp.requestHeaders &&
    (reqresp.requestHeaders["cookie"] || reqresp.requestHeaders["Cookie"]);
  if (hasCookie) {
    const cookie = reqresp.requestHeaders["cookie"] || reqresp.requestHeaders["Cookie"];
    httpHeaders["x-wabac-preset-cookie"] = cookie;
  }

```

Reply:
Oh but it does seem to work when testing, so maybe I'm mistaken! Could be I just don't understand properly what this is actually doing

Reply:
TIL something about Javascript 😆